src/app/ApplicationTemplate.Shared/Framework/Browser/DispatcherBrowserService.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using Windows.UI.Core;
+using Xamarin.Essentials;
+using Xamarin.Essentials.Interfaces;
+
+namespace ApplicationTemplate.Framework
+{
+	/// <summary>
+	/// Invokes calls to an IBrowser on the dispatcher
+	/// </summary>
+	public class DispatcherBrowserService : IBrowser
+	{
+		private readonly IBrowser _innerBrowser;
+		private readonly CoreDispatcher _dispatcher;
+
+		public DispatcherBrowserService(IBrowser browser, CoreDispatcher dispatcher)
+		{
+			_innerBrowser = browser;
+			_dispatcher = dispatcher;
+		}
+
+		public async Task OpenAsync(string uri)
+		{
+			await _dispatcher.RunAsync(CoreDispatcherPriority.Normal, () => _innerBrowser.OpenAsync(uri));
+		}
+
+		public async Task OpenAsync(string uri, BrowserLaunchMode launchMode)
+		{
+			await _dispatcher.RunAsync(CoreDispatcherPriority.Normal, () => _innerBrowser.OpenAsync(uri, launchMode));
+		}
+
+		public async Task OpenAsync(string uri, BrowserLaunchOptions options)
+		{
+			await _dispatcher.RunAsync(CoreDispatcherPriority.Normal, () => _innerBrowser.OpenAsync(uri, options));
+		}
+
+		public async Task OpenAsync(Uri uri)
+		{
+			await _dispatcher.RunAsync(CoreDispatcherPriority.Normal, () => _innerBrowser.OpenAsync(uri));
+		}
+
+		public async Task OpenAsync(Uri uri, BrowserLaunchMode launchMode)
+		{
+			await _dispatcher.RunAsync(CoreDispatcherPriority.Normal, () => _innerBrowser.OpenAsync(uri, launchMode));
+		}
+
+		public async Task<bool> OpenAsync(Uri uri, BrowserLaunchOptions options)
+		{
+			var success = false;
+			await _dispatcher.RunAsync(CoreDispatcherPriority.Normal, async () => success = await _innerBrowser.OpenAsync(uri, options));
```

Comment:
I'm not sure if there something equivalent in the project, but we could import those: https://github.com/nventive/Chinook.Navigation/blob/master/src/StackNavigation.Uno/Utils/Extensions/Windows.UI.Core.CoreDispatcher.cs#L46

Reply:
I feel like the calls are not properly awaited right now?